crates/net/network/src/manager.rs

```diff
@@ -842,20 +842,20 @@ impl<N: NetworkPrimitives> NetworkManager<N> {
                     "Session disconnected"
                 );
 
-                let mut reason = None;
-                if let Some(ref err) = error {
+                let reason = if let Some(ref err) = error {
```

Comment:
What are the benefits of this changes

Reply:
none really, it's just to make clippy happy, see [here](https://github.com/scroll-tech/reth/pull/273#discussion_r2212450336).